Week2/homework/js-exercises/GradeCalculater.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+
+//this function calculates grades
+function calculatesGrade(score){
+
+    if(score >= 0 && score <= 49 ){
+
+        console.log("You got a F (%s%)!",score);
+
+    }else if(score >= 50 && score <= 59){
+
+        console.log("You got a E (%s%)!",score);
+
+    }else if(score >= 60 && score <= 69){
+
+        console.log("You got a D (%s%)!",score);
+
+    }else if(score >= 70 && score <= 79){
+
+        console.log("You got a C (%s%)!",score);
+
+    }else if(score >= 80 && score <= 89){
+
+        console.log("You got a B (60%)!");
+
+    }else if(score >= 90 && score <= 100){
+
+        console.log("You got a A (%s%)!",score);
+    }
+
+}
+
+calculatesGrade(65);
+calculatesGrade(75);
+calculatesGrade(85);
```

Comment:
Well done, seems to be working just fine, on thing you could have noticed is rewriting the `console.log("You got a GRADE (%s%)!",score);` over and over again, usually this is what you want to prevent. 

So for instance what you could have done is:

```
let grade = null;

if(score >= 0 && score <= 49) {
    grade = 'F'
}
```

Then at the end of the function
```
console.log(`You got a ${ grade }!`, score);
```

Reply:
Another solution I created for one of your classmates is the following, maybe something more abstract but you might find it useful and helpful to improve your understanding.

```
const grades = ['A', 'B', 'C', 'D', 'E', 'F'].reverse();            // Reverse the array

function gradeCalculator(score) {
    const convert = Math.round((score - 50) * 0.1);                 // Convert score, for instance 100 turns to 5
    const grade = grades[convert] || 'F';                           // Select index from array or if that is not available return 'F' 
    return console.log(`You got a ${ grade } (${ score }%)!`);      // Return the score of the grade
}

gradeCalculator(78);
```
To function will return `"You got a C (78%)"`